<launch>
  <param name="urdf_file" textfile="$(find auv_estimator)/urdf/robot.urdf" />
  <arg name="bag_file_path" />
  
   <!-- GTSAM fusion combining IMU and 6DOF pose -->
  <node name="gtsam_fusion" pkg="auv_estimator" type="gtsam_fusion_ros.py" output="screen" required="true">
    <param name="bag_file_path" value="/home/allison/Workspace/AuvWs/src/auv_estimator/launch/CurvedTraj.bag" />
    <param name="pose_topic" value="/rexrov2/pose_gt" />
    <param name="imu_topic" value="/rexrov2/imu" />
    <param name="gps_interval" value="0.5" />
    <param name="use_pose" value="true" />
    <param name="use_gps" value="false" />
    <param name="plot_results" value="true" />
  </node>

  <!-- Robot state publisher resposible of publishing the static transforms -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher">
    <remap from="robot_description" to="urdf_file" />
  </node>
</launch>
